### PR TITLE
6791 vaults upgrade

### LINKED ISSUE #6791: make the VaultFactory contract upgradable
## What is the Problem Being Solved?

For the next release all contracts with precious state must be upgradable.

## Description of the Design

Adopt the vivification methods and iron out kinks.

One will be that outstandingQuote cannot survive upgrade because it's a promise and https://github.com/Agoric/agoric-sdk/issues/3787. Upon upgrade restart the liquidator activity, which is the loop waiting for the quote promise to be resolved. Basically, liquidationInProgress and outstandingQuote need to be made ephemeral.

## Test Plan

There must be an automated test of upgrade, using the new flow framework.

### PR DESCRIPTION
**Pull request by turadg**
closes: #6791 

## Description

Stacked on,
- https://github.com/Agoric/agoric-sdk/pull/7554

Makes the vaultFactory contract upgradable.

### Security Considerations

Governance params aren't yet durable so they will be reset upon contract restart. Any changes by the EC will have to be made again before service.

### Scaling Considerations

Some heap improvements. Only one singleton kind now (vaultManagers are now an Exo class kit)

### Documentation Considerations

--

### Testing Considerations

New bootstrap test. Careful review of test coverage requested. Anything missing?

### CODE REVIEW

File: packages/inter-protocol/src/vaultFactory/vaultDirector.js [hunk 1]
```diff
@@ -97,20 +96,21 @@ export const prepareVaultDirector = (
...
+  /** @type {import('../reserve/assetReserve.js').ShortfallReporter} */
+  let shortfallReporter;
+
   /** For holding newly minted tokens until transferred */
   const { zcfSeat: mintSeat } = zcf.makeEmptySeatKit();
 
   const rewardPoolSeat = provideEmptySeat(zcf, baggage, 'rewardPoolSeat');
 
-  /** @type {MapStore<Brand, VaultManager>} */
+  /** @type {MapStore<Brand, number>} */
```
Comment (Chris-Hibbert) [inline packages/inter-protocol/src/vaultFactory/vaultDirector.js]: please rename `collateralTypes`.

File: packages/inter-protocol/src/vaultFactory/vaultDirector.js [hunk 2]
```diff
@@ -187,15 +157,124 @@ export const prepareVaultDirector = (
...
+     * managers access to params from all managers. It's not POLA but it's a
+     * public authority and it reduces the number of distinct power objects to
+     * create.
+     *
+     * @param {Brand} brand
+     */
+    getGovernedParams: brand => {
+      const vaultParamManager = vaultParamManagers.get(brand);
+      return Far('vault manager param manager', {
+        // merge direction and manager params
```
Comment (Chris-Hibbert) [inline packages/inter-protocol/src/vaultFactory/vaultDirector.js]:
```suggestion
        // merge director and manager params
```

File: packages/inter-protocol/src/vaultFactory/vaultFactory.js [hunk 3]
```diff
@@ -62,7 +66,12 @@ export const start = async (zcf, privateArgs, baggage) => {
     marshaller,
     storageNode,
   });
-  const debtMint = await zcf.registerFeeMint('Minted', feeMintAccess);
+
+  trace('awaiting debtMint');
```
Comment (Chris-Hibbert) [inline packages/inter-protocol/src/vaultFactory/vaultFactory.js:70]: Were these new `trace()` calls intended to be included?  [fine if true, just wanted to post a reminder in case they were for short-term debugging purposes.]

File: packages/zoe/src/contractSupport/durability.js [hunk 4]
```diff
@@ -53,34 +49,80 @@ harden(provideEmptySeat);
...
  * @param {import('@agoric/vat-data').Baggage} baggage
  * @param {string} category diagnostic tag
+ * @param {M} prepareSingleton
+ * @param {{}} categoryPowers
  */
-export const provideChildBaggage = (baggage, category) => {
-  const baggageSet = provideDurableSetStore(baggage, `${category}Set`);
-  return Far('childBaggageManager', {
-    // TODO(types) infer args
+export const provideCategorySingletons = (
```
Comment (Chris-Hibbert) [inline packages/zoe/src/contractSupport/durability.js]: This seems unrelated to the rest of the PR. `provideCategorySingletons` and `makeSingletons` are defined, but not used here. Have I missed something?

File: packages/vats/test/bootstrapTests/test-vaults-upgrade.js [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+  const vfAdminFacet = await EV(
+    vaultFactoryKit.governorCreatorFacet,
+  ).getAdminFacet();
+
+  const keyMetrics = {
+    numActiveVaults: 1,
+    totalCollateral: { value: 9000000n },
+    totalDebt: { value: 5025000n },
+  };
+  // as
```
Comment (Chris-Hibbert) [inline packages/vats/test/bootstrapTests/test-vaults-upgrade.js]: What does `as` mean here?

File: packages/vats/test/bootstrapTests/test-vaults-upgrade.js [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+    walletFactoryDriver,
+  };
+};
+
+test.before(async t => {
+  t.context = await makeDefaultTestContext(t);
+});
+test.after.always(t => t.context.shutdown());
+
+test.serial('open vault', async t => {
```
Comment (Chris-Hibbert) [inline packages/vats/test/bootstrapTests/test-vaults-upgrade.js:72]: does `test.serial` that they're run in the order in which they appear in the file?

**Review by Chris-Hibbert — CHANGES_REQUESTED** [review]

File: packages/zoe/src/contractSupport/durability.js (hunk 4 above)
Comment (turadg) [reply packages/zoe/src/contractSupport/durability.js]: Remnant of an interim state. Forgot to drop them

File: packages/vats/test/bootstrapTests/test-vaults-upgrade.js (hunk 5 above)
Comment (turadg) [reply packages/vats/test/bootstrapTests/test-vaults-upgrade.js]: Stray characters

File: packages/vats/test/bootstrapTests/test-vaults-upgrade.js (hunk 6 above)
Comment (turadg) [reply packages/vats/test/bootstrapTests/test-vaults-upgrade.js:72]: Correct! https://github.com/avajs/ava/blob/main/docs/01-writing-tests.md#running-tests-serially

File: packages/inter-protocol/src/vaultFactory/vaultDirector.js (hunk 1 above)
Comment (turadg) [reply packages/inter-protocol/src/vaultFactory/vaultDirector.js]: thx done

File: packages/inter-protocol/src/vaultFactory/vaultDirector.js (hunk 2 above)
Comment (turadg) [reply packages/inter-protocol/src/vaultFactory/vaultDirector.js]: done

File: packages/inter-protocol/src/vaultFactory/vaultFactory.js (hunk 3 above)
Comment (turadg) [reply packages/inter-protocol/src/vaultFactory/vaultFactory.js:70]: Yes, but I don't the tracer on. disabled

**Review by Chris-Hibbert — APPROVED** [review]
